v2/eventsource/stores/sqlstore/store.go

```diff
@@ -22,10 +29,18 @@ var (
 	loadSQL = "SELECT aggregate_id, sequence_id, created_at, user_id, type, data FROM %s"
 )
 
-// New creates a new event source store
+// New creates a new event source store.
 func New(db *sql.DB, tableName string) eventsource.Store {
 	return &store{
-		db:        db,
+		db:        &driver.Generic{DB: db},
+		tablename: tableName,
+	}
+}
+
+// New creates a new event source store.
+func NewPgx(db *pgxpool.Pool, tableName string) eventsource.Store {
```

Comment:
Ideas of nicer ways to do this are welcome. (I guess `func New(db *sql.DB, tableName string) eventsource.Store {` needs to stay unless we want to make a V3.0)

Reply:
I think we should rethink this library for v3, there is a lot of oddities that should be changed imo.

Regarding the interface, do the two structs share something in common that we can create an interface for? Otherwise, I think this is the best solution.

Reply:
All the methods in pq and pgx have different signatures (pgx has context everywhere), so I don't think a common interface is viable.